### DISCUSSION TITLE
LAB 03.02 task 2

### DISCUSSION THREAD

**Discussion by jaayg23** [rramosp/ai4eng.v1#161] [03 - GENERACION DE MODELOS]
Buen día, para la tarea 2 he implementado el siguiente código para la solución de la tarea 2.
`r = np.dot(np.insert(np.array(td.loc[:, td.columns!='signal+1']),0, 1, axis=1),w)`
Pero me retorna una solución con alguna diferencia en las decimas del vector, seguramente el error estará en la forma en como implemento un vector de unos en la matriz inicial para poder hacer el .dot, pero no entiendo la razón de este problema si estoy multiplicando el primer w por 1.
Muchas gracias

**Comment by Felipe-RA** [comment]
Saludos jaayg23

Esa pequeña diferencia ocurre porque tu línea de código no se traduce a una aplicación correcta de la regresión:

$$\hat{y} = w_0 + w_1s + w_2s_{-1} + w_3s_{-2} + w_4s_{-3} ... $$ 

¿Qué parte de esta función no tiene en cuenta tu implementación?

Para lograr encontrar el problema te invito a implementar una solución primero de forma desglosada (con `ciclos for` y `if`) para comprender lo que te falta en tu implementación actual.

Una vez tengas esto puedes dar el paso a cumplir el reto de implementarlo en una sola línea.

Te invito a que te hagas las siguientes preguntas para solucionar el problema:

1. ¿Qué debo iterar?
2. ¿Qué caso(s) particulares existen en cada iteración?
3. ¿Cómo debo tratar estos casos particulares?

Si necesitas más ayuda, por aquí estaremos! Ya estas muy cerca!

PD: ¡Disculpa la demora en responder tu post!
